src/components/App/App.js

```diff
@@ -1,65 +1,320 @@
+      );
+
+      setSearchResults(updatedSearchResults);
+    };
+
+    try {
+      if (cardToDelete && cardToDelete._id) {
+        await removeSavedNews(cardToDelete._id, token);
+        updateNews({ ...newsData, _id: "", isSaved: false }, true);
+        // console.log("Article removed:", cardToDelete);
+      } else {
+        console.error("Failed to find article to delete:", cardToDelete);
+        throw new Error("Failed to find article to delete");
+      }
+    } catch (err) {
+      console.log(err);
+    }
+  };
+
+  useEffect(() => {
+    const token = localStorage.getItem("token");
+    if (token) {
+      checkToken(token)
+        .then((res) => {
+          setCurrentUser(res);
+          setIsLoggedIn(true);
+        })
+        .catch(console.error)
+        .finally(() => setIsLoggedInLoading(false));
+    } else {
+      setIsLoggedInLoading(false);
+    }
+  }, []);
+
   return (
-    <div className="app">
-      <Routes>
-        <Route exact path="/" element={<Main onLogin={openLoginModal} />} />
-        <Route exact path="/saved-news" element={<SavedNews />} />
-      </Routes>
-
-      <Footer />
-      {activeModal === "signup" && (
-        <RegisterModal
-          handleCloseModal={handleCloseModal}
-          isOpen
-          switchToLogin={handleRedirect}
-          isLoading={isLoading}
-        />
-      )}
-      {activeModal === "login" && (
-        <LoginModal
-          handleCloseModal={handleCloseModal}
-          isOpen
-          switchToRegister={handleRedirect}
-          isLoading={isLoading}
-        />
-      )}
-    </div>
+    <CurrentPageContext.Provider value={{ currentPage, setCurrentPage }}>
+      <CurrentUserContext.Provider value={{ currentUser, setCurrentUser }}>
+        <SavedNewsContext.Provider value={{ savedNews, setSavedNews }}>
+          <HasSearchedContext.Provider value={{ hasSearched }}>
+            <SearchResultContext.Provider value={{ searchResults }}>
+              <KeywordContext.Provider value={{ keyword, setKeyword }}>
+                <div className="app">
+                  <Routes>
+                    <Route
+                      exact
+                      path="/"
+                      element={
+                        <Main
+                          onLogin={openLoginModal}
+                          onLogout={handleLogout}
+                          isLoggedIn={isLoggedIn}
+                          isLoading={isSearching}
+                          handleSearch={handleSearch}
+                          searchError={searchError}
+                          handleSaveCard={handleSaveCard}
+                          handleDeleteCard={handleDeleteCard}
+                          handleMobileModal={openMobileModal}
+                        />
+                      }
+                    />
+                    <Route
+                      exact
+                      path="/saved-news"
+                      element={
+                        <SavedNews
```

Comment:
* I can visit `Saved acticles` even if I'm not authorized https://snipboard.io/23gqsO.jpg . But it should be **protected**

Please, use `ProtectedRoute` for it